src/location.rs

```diff
@@ -153,42 +155,75 @@ pub fn wait_for_location(
     let root = screen.root();
     let preview_width = preview_width.ensure_odd();
 
-    let pointer = xproto::query_pointer(conn, root).get_reply()?;
-    let pointer_pos = (pointer.root_x(), pointer.root_y());
-    let (width, initial_rect) =
-        get_window_rect_around_pointer(conn, screen, pointer_pos, preview_width, scale)?;
+    let mut curr_scale = scale;
+    let mut curr_preview_width = preview_width;
+
+    macro_rules! create_new_cursor {
```

Comment:
Personally, I would maybe replace this macro with two separate functions. Macros are nice but might not reach for them just to simulate functions with variable arity. Of course this is somewhat subjective.

As a side note I am a bit concerned if `create_new_cursor` currently leaks resources. Should the returned cursor ID be freed after it is no longer needed?

Reply:
> Personally, I would maybe replace this macro with two separate functions. Macros are nice but might not reach for them just to simulate functions with variable arity. Of course this is somewhat subjective.

Yeah no worries, I picked a macro because I didn't like having two functions that had almost the same code. I've refactored it out to use a single function now that takes an `Option<(i16, i16)>`. Let me know if that works for you.

> As a side note I am a bit concerned if create_new_cursor currently leaks resources. Should the returned cursor ID be freed after it is no longer needed?

Oh, you're right! I'd totally missed that :sweat_smile: I've added in the `xproto::free_cursor` calls now. :slightly_smiling_face: